inference-engine/src/transformations/src/transformations/common_optimizations/compress_float_constants.cpp

```diff
@@ -0,0 +1,119 @@
+        } else {
+            dst_data[i] = static_cast<dst_type>(src_data[i]);
+        }
+    }
+    return new_constant;
+}
+}  // namespace
+
+ov::pass::CompressFloatConstantsImpl::CompressFloatConstantsImpl() {
+    MATCHER_SCOPE(CompressFloatConstantsImpl);
+    auto const_pattern = pattern::wrap_type<opset8::Constant>();
+
+    ov::matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        const auto& pattern_map = m.get_pattern_value_map();
+        const auto& const_node_pattern = pattern_map.at(const_pattern);
+
+        auto const_node = std::dynamic_pointer_cast<ov::opset8::Constant>(
+            const_node_pattern.get_node_shared_ptr());
+        if (!const_node)
+            return false;
+
+        if (ov::fp16_compression_is_disabled(const_node))
+            return false;
+
+        auto c_type = const_node->get_element_type();
+        std::shared_ptr<ov::Node> new_const;
+        if (c_type == ov::element::f32) {
+            new_const = change_constant_precision<ov::element::Type_t::f32, ov::element::Type_t::f16>(const_node);
+        } else if (c_type == ov::element::f64) {
+            new_const = change_constant_precision<ov::element::Type_t::f64, ov::element::Type_t::f16>(const_node);
+        } else {
+            return false;
+        }
+        auto convert = std::make_shared<ov::opset8::Convert>(new_const, const_node->get_element_type());
+
+        convert->set_friendly_name(const_node->get_friendly_name());
+        ngraph::copy_runtime_info(const_node, convert);
+        ov::mark_as_decompression(convert);
+
+        ov::replace_node(const_node, convert);
+
+        return true;
+    };
+
+    auto m = std::make_shared<pattern::Matcher>(const_pattern, matcher_name);
+    this->register_matcher(m, callback);
+}
+
+ov::pass::AddOldApiMapToParameters::AddOldApiMapToParameters() {
+    MATCHER_SCOPE(AddOldApiMapToParameters);
+    auto param_pattern = pattern::wrap_type<opset8::Parameter>();
+
+    ov::matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        const auto& pattern_map = m.get_pattern_value_map();
+        auto node = pattern_map.at(param_pattern).get_node_shared_ptr();
+
+        auto param_node = std::dynamic_pointer_cast<ov::opset8::Parameter>(node);
+        if (!param_node)
+            return false;
+        auto p_type = param_node->get_element_type();
+        if (p_type == ov::element::f32 || p_type == ov::element::f64) {
+            std::vector<uint64_t> order;
+            if (ov::has_old_api_map(node)) {
+                auto old_api = ov::get_old_api_map(node).get();
+                order = old_api.get_order();
+            } else {
+                auto p_rank = param_node->get_partial_shape().rank();
+                if (p_rank.is_static()) {
+                    auto r = p_rank.get_length();
+                    order.resize(r);
+                    std::iota(order.begin(), order.end(), 0);
+                } else {
+                    return false;
```

Comment:
why can't we handle dynamic rank and use {} for order?

Reply:
old api map is used for old api, right? Old api doesnt support dynamic rank anyway

Reply:
This is open question. It makes sense to discuss offline how we handle dynamic rank paramters.